### ISSUE TITLE
Extract clientId from ServiceAccount

### ISSUE THREAD

**Issue by UXabre** [Azure/secrets-store-csi-driver-provider-azure#1080]
**Describe the solution you'd like**
In the AAD Pod ID Identity days, I only needed to say that I'd like to use MSI and everything else simply works. Now I have to add duplicate data, namely the ClientID to the secretsprovider as well as the serviceaccount.
I'd like that there's an explicit flag for workloadIdentity and that the clientId is extracted from the service account

**Anything else you would like to add:**
The reason is as such: we're using ArgoCD + Terraform. Terraform sets up the azure ground layer, like the AKS cluster and also the identities that need to be created. Now I need to find a way to add runtime data (a clientId can change if my terraform script would find a good reason to replace the identity). In a gitops fashion, that would mean that I have to now do two updates in my code.
What's more, what if I'd like to reuse my secretsprovider? It is hardlinked to an identity now which makes it far less (re)usable as I know can only use a single identity to expose multiple things... in the past I could use different identities for different use-cases

**Environment:**

- Secrets Store CSI Driver version: v1.3.0
- Azure Key Vault provider version: (use the image tag): v1.4.0
- Kubernetes version: (use `kubectl version`): 1.24.9
- Cluster type: (e.g. AKS, aks-engine, etc): AKS

**Comment by UXabre** [comment]
Closing this because PEBKAC... I needed to change my way i think about identity lifecycles in my project. But creating it out of band of my script it's easy to implement

**Comment by calebedwards21** [comment]
@UXabre We are running into some of the same issues and are wondering if you could explain a little more what you did?

**Comment by UXabre** [comment]
Hi @calebedwards21 , thanks for reaching out!
I have it set up like this:
- a small bash script sets up the identities using az cli (this is brw what i meant by "creating it out of band", as these now live completely out of my terraform state
- the clientid is extracted into a json that i store in my git repository
- i use this data to inject in my terraform, e.g. to set up federation
- this json is also used in argocd, as i use ApplicationSet, it has the ability to extract data from a json file and use it to generate values for my helm charts

Its kinda straightforward i think and perhaps there are better ways, if you know any: plz let me know 😓 fpr now this seems like an easy way out.

I hope this information helps you out!

**Comment by calebedwards21** [comment]
@UXabre So are you using a separate SecretProviderClass for every Deployment, Job, etc? You're not reusing that SecretProviderClass?

**Comment by UXabre** [comment]
Yes, I'm using a separate secretproviderclass for where i need it. Full disclosure though is that I tend to only use this to setup hashicorp vault and those integrations work much more seamless with k8s and ...you know... Simply anything. Makes it possible to have more fine grained policies and links easy to serviceacvounts but might be overkill for your use case. 
So: yes, separate SPCs but I don't create that many in my entire infra.
I do think that is a step down from the first proposal but I don't think I'll be able to change anyone's mind on the topic :p

**Comment by calebedwards21** [comment]
Great, thanks for the quick responses! I like your initial proposal and was hoping there was something out there for it
